scapy/layers/geneve.py

```diff
@@ -0,0 +1,50 @@
+#! /usr/bin/env python
+# (GENEVE):
+# A Framework for Overlaying Virtualized Layer 2 Networks over Layer 3 Networks
+
+from scapy.packet import Packet, bind_layers
+from scapy.layers.l2 import Ether
+from scapy.layers.inet import IP, UDP
+from scapy.layers.inet6 import IPv6
+from scapy.fields import FlagsField, XByteField, ThreeBytesField, \
+    ConditionalField, ShortField, ByteEnumField, X3BytesField
+
+#
+# +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+# |Ver|  Opt Len  |O|C|    Rsvd.  |          Protocol Type        |
+# +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+# |        Virtual Network Identifier (VNI)       |    Reserved   |
+# +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+# |                    Variable Length Options                    |
+# +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+#
+
+class GENEVE(Packet):
+    name = "GENEVE"
+
+    fields_desc = [
+        # INIT = ver + optlen + o + c + rsvd (all zeros)
+        ShortField("init", 0x0),
```

Comment:
IMHO, it would be more helpful to separately define the five fields in the first 16 bit short, instead of lumping them into one 16 bit field.